test/test_circular_buffer.jl

```diff
@@ -73,4 +73,9 @@
     end
     @test isempty(cb)
     @test_throws ArgumentError popfirst!(cb)
+    
+    # test fill!(cb, data)
```

Comment:
I recommend when adding new tests to a file in DataStructures,
always use nested `@testsets` per functionality
and  taking the opportunity to update the remaining earlier tests to use testsets too.

The only reason it isn't that way, is because DataStructures.jl is much older than `@testset`

So this would become 

```
@testset "fill!" begin
    cb = CircularBuffer{Int}(3)
    fill!(cb, 42)
    @test Array(cb) == [42, 42, 42]
end
```